test/replutil.jl

```diff
@@ -134,6 +144,81 @@ let
     showerror(buff, MethodError(convert, Tuple{DataType, Float64}))
 end
 
+# Issue #14884
+bitstype 8 EightBitType
+bitstype 8 EightBitTypeT{T}
+immutable FunctionLike <: Function; end
+let err_str,
+    i = reinterpret(EightBitType, 0x54),
+    j = reinterpret(EightBitTypeT{Int32}, 0x54)
+
+    err_str = @except_str Symbol() MethodError
+    @test contains(err_str, "no method matching Symbol()")
+    err_str = @except_str :a() MethodError
+    @test contains(err_str, "no method matching (::Symbol)()")
+    err_str = @except_str EightBitType() MethodError
+    @test contains(err_str, "no method matching EightBitType()")
+    err_str = @except_str i() MethodError
+    @test contains(err_str, "no method matching (::EightBitType)()")
+    err_str = @except_str EightBitTypeT() MethodError
+    @test contains(err_str, "no method matching EightBitTypeT{T}()")
+    err_str = @except_str EightBitTypeT{Int32}() MethodError
+    @test contains(err_str, "no method matching EightBitTypeT{Int32}()")
+    err_str = @except_str j() MethodError
+    @test contains(err_str, "no method matching (::EightBitTypeT{Int32})()")
+    err_str = @except_str FunctionLike()() MethodError
+    @test contains(err_str, "no method matching (::FunctionLike)()")
+end
+@test stringmime("text/plain", FunctionLike()) == "(::FunctionLike) (generic function with 0 methods)"
+@test ismatch(r"@doc \(macro with \d+ method[s]?\)", stringmime("text/plain", Base.(symbol("@doc"))))
+
+method_defs_lineno = @__LINE__
+Base.Symbol() = error(1)
+(::Symbol)() = error(2)
```

Comment:
if it comes up, we can add a dummy argument (perhaps EightBitType). but for now, i specifically want to test Symbol since it has a special and unique representation.